kafka-ui-react-app/src/components/Topics/Details/Messages/MessagesTable.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { TopicMessage } from 'generated-sources';
+import CustomParamButton, {
+  CustomParamButtonType,
+} from '../../shared/Form/CustomParams/CustomParamButton';
+import MessageItem from './MessageItem';
+
+interface MessagesTableProp {
+  messages: TopicMessage[];
+  onNext(event: React.MouseEvent<HTMLButtonElement>): void;
+}
+
+const MessagesTable: React.FC<MessagesTableProp> = ({ messages, onNext }) => {
+  if (!messages.length) {
+    return <div>No messages at selected topic</div>;
+  }
+
+  return (
+    <div>
+      <table className="table is-striped is-fullwidth">
+        <thead>
+          <tr>
+            <th>Timestamp</th>
+            <th>Offset</th>
+            <th>Partition</th>
+            <th>Content</th>
+          </tr>
+        </thead>
+        <tbody>
+          {messages.map(
+            ({ partition, offset, timestamp, content }: TopicMessage) => (
```

Comment:
I would suggest do not do spreading`TopicMessage` outside and just provide it to `MessageItem` as a single prop like `topic` or `topicMessage`.